runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -32,14 +31,20 @@
     @Getter
     byte[] data;
 
-    private ByteBuf serializedCache = null;
+    private SerializedCache serializedCache = null;
 
     private int lastKnownSize = NOT_KNOWN;
 
     private final transient AtomicReference<Object> payload = new AtomicReference<>();
 
     private final EnumMap<LogUnitMetadataType, Object> metadataMap;
 
+    @RequiredArgsConstructor
+    private class SerializedCache {
```

Comment:
it should be static

![image](https://user-images.githubusercontent.com/314292/89974193-392f9680-dc17-11ea-8012-90297b9ed463.png)

take into account how many instances of LogData we have when Corfu works it should be a huge overhead

Reply:
What type of overhead ?

Reply:
Done.

Reply:
> What type of overhead ?

This is what I remember from "Effective Java" 
https://stackoverflow.com/questions/43998428/why-to-favor-static-member-classes-over-nonstatic-joshua-bloch-item-22#:~:text=Answer%3A,reference%20to%20its%20enclosing%20instance.